lua/entities/gmod_wire_expression2/core/custom/sv_vex_main.lua

```diff
@@ -7,54 +7,38 @@
  Random Misc. Functions that are cool like hiding other people's chat (probably doesn't work) and setting the ranger Filter.                    
 ]]
 
-local E2Table = function() return {n={},ntypes={},s={},stypes={},size=0} end
+local newE2Table, luaTableToE2 = vex.newE2Table, vex.luaTableToE2
+local table_insert = table.insert -- Gonna be using this a lot
 
-local function luaTablToE2(T)
-    local Strt = E2Table()
-    local Sz = 0
-    for Key,Value in pairs(T) do
-        local TypeV = type(Value)
-        local WriteV = Strt.n
-        local WriteType = Strt.ntypes
-        if type(Key)=="string" then WriteV = Strt.s WriteType=Strt.stypes end
-        local Clean = Value
-        if TypeV=="bool" then Clean = Value and 1 or 0 elseif
-        TypeV=="table" then Clean = luaTablToE2(Value) end
-        Sz = Sz + 1
-        WriteV[Key] = Clean
-        WriteType[Key] = TypeV[1]
+-- Takes out everything that isn't a valid entity in a table
+local function cleanupTable(t)
+    local out = {}
+    for _,to_filter in pairs(filter) do
+        if isentity(to_filter) then
+            table_insert(fixed,to_filter)
+        end
     end
-    Strt.size = Sz
-    return Strt
+    return out
 end
 
 -- Lib functions
 
 __e2setcost(10)
 e2function table rangerOffsetManual(vector pos,vector endpos, array filt)
-    local Start = Vector(pos[1],pos[2],pos[3])
-    local End = Vector(endpos[1],endpos[2],endpos[3])
-	local tr = util.TraceLine( {
-		start = Start,
-		endpos = End,
-		filter = filt
-    } )
-	if not tr then return E2Table() end
-	return luaTablToE2(tr)
+    local tr = util.TraceLine {
+        start = Vector(pos[1],pos[2],pos[3]),
+        endpos = Vector(endpos[1],endpos[2],endpos[3]),
+        filter = cleanupTable(filt)
+    }
+    return tr and luaTableToE2(tr) or newE2Table()
```

Comment:
If you do add type-guessing for `ranger`, then this E2 function should be changed to return `ranger` instead of `table`.